ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/services/grpc/GrpcCommandService.scala

```diff
@@ -39,58 +40,30 @@ class GrpcCommandService(
     FieldValidations(ErrorFactories(errorCodesVersionSwitcher)),
   )
 
-  override def submitAndWait(request: SubmitAndWaitRequest): Future[Empty] = {
-    val requestWithSubmissionId = generateSubmissionIdIfEmpty(request)
-    validator
-      .validate(
-        requestWithSubmissionId,
-        currentLedgerTime(),
-        currentUtcTime(),
-        maxDeduplicationTime(),
-      )(contextualizedErrorLogger(requestWithSubmissionId))
-      .fold(
-        t => Future.failed(ValidationLogger.logFailure(requestWithSubmissionId, t)),
-        _ => service.submitAndWait(requestWithSubmissionId),
-      )
-  }
+  override def submitAndWait(request: SubmitAndWaitRequest): Future[Empty] =
+    enrichRequestAndSubmit(request)(service.submitAndWait)
 
   override def submitAndWaitForTransactionId(
       request: SubmitAndWaitRequest
-  ): Future[SubmitAndWaitForTransactionIdResponse] = {
-    val requestWithSubmissionId = generateSubmissionIdIfEmpty(request)
-    validator
-      .validate(
-        requestWithSubmissionId,
-        currentLedgerTime(),
-        currentUtcTime(),
-        maxDeduplicationTime(),
-      )(contextualizedErrorLogger(requestWithSubmissionId))
-      .fold(
-        t => Future.failed(ValidationLogger.logFailure(requestWithSubmissionId, t)),
-        _ => service.submitAndWaitForTransactionId(requestWithSubmissionId),
-      )
-  }
+  ): Future[SubmitAndWaitForTransactionIdResponse] =
+    enrichRequestAndSubmit(request)(service.submitAndWaitForTransactionId)
 
   override def submitAndWaitForTransaction(
       request: SubmitAndWaitRequest
-  ): Future[SubmitAndWaitForTransactionResponse] = {
-    val requestWithSubmissionId = generateSubmissionIdIfEmpty(request)
-    validator
-      .validate(
-        requestWithSubmissionId,
-        currentLedgerTime(),
-        currentUtcTime(),
-        maxDeduplicationTime(),
-      )(contextualizedErrorLogger(requestWithSubmissionId))
-      .fold(
-        t => Future.failed(ValidationLogger.logFailure(requestWithSubmissionId, t)),
-        _ => service.submitAndWaitForTransaction(requestWithSubmissionId),
-      )
-  }
+  ): Future[SubmitAndWaitForTransactionResponse] =
+    enrichRequestAndSubmit(request)(service.submitAndWaitForTransaction)
 
   override def submitAndWaitForTransactionTree(
       request: SubmitAndWaitRequest
-  ): Future[SubmitAndWaitForTransactionTreeResponse] = {
+  ): Future[SubmitAndWaitForTransactionTreeResponse] =
+    enrichRequestAndSubmit(request)(service.submitAndWaitForTransactionTree)
+
+  override def bindService(): ServerServiceDefinition =
+    CommandServiceGrpc.bindService(this, executionContext)
+
```

Comment:
It only looks as being moved because of the public/private sorting. I would prefer to have it as the last method as it's not really relevant. Do you prefer having it after the private method?